### PR TITLE
[Draft,to-be-split] 4 new hdf5-vols packages

### PR DESCRIPTION
**Pull request by lrknox**
Added packages hdf5-vol-async, hdf5-vol-cache and hdf5-vol-log.

### CODE REVIEW

File: var/spack/repos/builtin/packages/hdf5-vol-cache/package.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    """Package for HDF5 cache VOL."""
+
+    homepage = "https://sdm.lbl.gov/"
+    git      = "https://github.com/hpc-io/vol-cache.git"
+
+    maintainers = ['hyoklee']
+
+    version('default', branch='develop')
+
+    # Set hdf5-cmake package option.
```
Comment (hyoklee) [inline var/spack/repos/builtin/packages/hdf5-vol-cache/package.py:18]: Remove the 5 lines of comment block.

File: var/spack/repos/builtin/packages/hdf5-vol-async/package.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+
+class Hdf5VolAsync(CMakePackage):
+    """This package enables asynchronous IO in HDF5."""
+
+    homepage = "https://sdm.lbl.gov/"
+    git      = "https://github.com/hpc-io/vol-async"
+    maintainers = ['hyoklee']
+
+    version('develop', branch='develop')
```
Comment (hyoklee) [inline var/spack/repos/builtin/packages/hdf5-vol-async/package.py:16]: Add a different version until https://github.com/hpc-io/vol-async/pull/13 is accepted.
```
    version('develop', branch='develop')
    version('hyoklee.develop',
            branch='develop',
            git='https://github.com/hyoklee/vol-async', preferred=True)
```

**Review by hyoklee — DISMISSED** [review]

**Comment by lrknox** [comment]
There were 2 PR checks that failed before – one was an ubuntu download, the other a test in an e4s pipeline – it wasn’t clear from a quick scan of the log whether it had anything to do with the PR.

The e4s pipeline is not required for the PR, the ubuntu test is.

The tests are queued and waiting to run.  I’m going to click the Ready for review button and see what happens.

Hmmm, it says “Merging can be performed automatically with 1 approving review.”
But also “You’re not authorized to merge this pull request.”
Taking both under consideration – I don’t know what that means.  Maybe automatically is someone with sufficient privilege clicking the “Merge pull request” button.

Anyway, it might be a while before these tests run.

**Review by hyoklee — APPROVED** [review]

**Review by bernhardkaindl — CHANGES_REQUESTED** [review]
@lrknox @hyoklee I, and I assume other maintainers would have too, have several issues with this PR.
* Just to mention it: You are submitting 4, not 3 packages as written with a single PR.
* So far, you needed 11 commits changing these 4 packages, not in an organized way.
* You are using only using develop branches, no stable branches.
* You are applying a patch on top of the develop branch for one of these packages
```
    version('default', branch='develop')
    depends_on('hdf5@develop-1.13')
    patch('CMakeLists.patch',
         sha256='09876179329cbc7d925055c714cabe05511ba7a9a8e9d5e029cc9acfdac5ffa7')
```
* The review comments by @hyoklee have not been answered and not fixed.
* There are too many not helpful blank lines in the package.py files in my opinion.

Reviews like this are too large to be effective and manageable for a reviewer.

Small PRs are the norm, one PR per package. But don't open 4 new PRs just yet:

I propose this plan:

1. Open one PR for hdf5-vol-async/package.py
2. Work on getting hdf5-vol-external-passthrough/CMakeLists.patch upstream.
3. While 1 and 2 are in progess, check this
```
grep -r depends_on.*hdf5 hdf5-vol-*hdf5-vol-async/package.py:    depends_on('hdf5@develop-1.13+mpi+threadsafe')
hdf5-vol-cache/package.py:    depends_on('hdf5-vol-async')
hdf5-vol-external-passthrough/package.py:    depends_on('hdf5@develop-1.13')
hdf5-vol-log/package.py:    depends_on('hdf5@1.12.1')
```
Do you really want 3 different variants/versions of hdf5 for this group of packages?
Besides prolonged compilation time, I am sure how it is the use different versions.

hdf5-vol-async needs to be merged first, as vol-cache depends on it.
vol-cache should only be submitted whith vol-async merged into spack.

**Comment by lrknox** [comment]
Thanks, @bernhardkaindl.  Step 1 of your proposed plan is done in PR #26874, using stable version tags instead of develop.
The upstream CMakeLists.txt patch has been merged, and another PR for hdf5-vol-external-passthrough will follow soon, probably with the same version tags.
hdf5-vol-cache will require a different tag which is not yet available, but when it is all three packages should be updated to use the new tag.
hdf5-vol-log may or may not be updated from hdf5 version 1.12.1.  At present it will not build with the 1.13.0-rc6 tag.

**Comment by alalazo** [comment]
Closing, since there has been no activity in a while. Feel free to reopen if you want to continue work on this.

**Pull request closed without merging**